core/accounting/src/chart_of_accounts/ledger/mod.rs

```diff
@@ -316,4 +321,203 @@ impl ChartLedger {
+            .await?;
+        let expenses_account_balances = self
+            .cala
+            .balances()
+            .effective()
+            .find_all_in_range(&expense_accounts, from, Some(until))
+            .await?;
+
+        Ok(ClosingAccountBalances {
+            revenue: revenue_account_balances,
+            cost_of_revenue: cost_of_revenue_account_balances,
+            expenses: expenses_account_balances,
+        })
+    }
+
+    async fn find_all_accounts_by_parent_set_id(
+        &self,
+        parent_set_id: AccountSetId,
+    ) -> Result<Vec<BalanceId>, ChartLedgerError> {
+        let mut accounts: Vec<BalanceId> = Vec::new();
+
+        let mut has_next_page = true;
+        let mut after = None;
+
+        while has_next_page {
+            let PaginatedQueryRet {
+                entities,
+                has_next_page: next_page,
+                end_cursor,
+            } = self
+                .cala
+                .account_sets()
+                .list_members_by_created_at(parent_set_id, PaginatedQueryArgs { first: 100, after })
+                .await?;
+
+            after = end_cursor;
+            has_next_page = next_page;
+
+            for member in entities {
+                match member.id {
+                    AccountSetMemberId::Account(account_id) => {
+                        accounts.push((self.journal_id, account_id, Currency::USD));
+                    }
+                    AccountSetMemberId::AccountSet(account_set_id) => {
+                        let nested_accounts =
+                            Box::pin(self.find_all_accounts_by_parent_set_id(account_set_id))
+                                .await?;
+                        accounts.extend(nested_accounts);
+                    }
+                }
+            }
+        }
+        Ok(accounts)
+    }
+
+    async fn create_equity_entry(
+        &self,
+        op: &mut LedgerOperation<'_>,
+        name: String,
+        equity_retained_earnings_account_set_id: AccountSetId,
+        equity_retained_losses_account_set_id: AccountSetId,
+        net_earnings: Decimal,
+    ) -> Result<ClosingAccountEntry, ChartLedgerError> {
+        let account = if net_earnings >= Decimal::ZERO {
+            self.create_account_in_op(
+                op,
+                name,
+                DebitOrCredit::Credit,
+                equity_retained_earnings_account_set_id,
+            )
+            .await?
+        } else {
+            self.create_account_in_op(
+                op,
+                name,
+                DebitOrCredit::Debit,
+                equity_retained_losses_account_set_id,
+            )
+            .await?
+        };
+
+        Ok(ClosingAccountEntry {
+            account_id: account.id.into(),
+            currency: Currency::USD,
+            amount: net_earnings.abs(),
+            direction: account.values().normal_balance_type,
+        })
+    }
+
+    async fn create_account_in_op(
```

Comment:
The name confused me a bit at first at call site

```suggestion
    async fn create_child_account_in_op(
```